src/core/domain-services/remote-key-service.ts

```diff
@@ -0,0 +1,196 @@
+            const data = await this.handleMessage(msg);
+            console.log("Inside RemoteKeyHost::initialize::Data(handleMessage): ", data);
+            this.sendInvocationResult(data, senderId!);
+        });
+    }
+    private async sendInvocationResult(result: any, receiverId: CruxId) {
+        if (!this.cruxProtocolMessenger) {
+            throw Error("RemoteKeyClient cannot send with no selfMessenger");
+        }
+        const resultData = this.generateInvocationResponse(result);
+        console.log("RemoteKeyHost::Inside sendInvocationResult::resultData: ", resultData);
+        // @ts-ignore
+        await this.cruxProtocolMessenger.send({
+            content: resultData,
+            type: "KEY_MANAGER_RESPONSE",
+        }, receiverId);
+    }
+
+    private async handleMessage(message: any) {
+        if (!VALID_METHODS.includes(message.method)) {
+            throw new Error("Invalid key manager method");
+        }
+        if (!this.keyManager) {
+            throw new Error("Key Manager not available");
+        }
+        let data;
+        if (message.method === "signWebToken") {
+            console.log("HandleMessage entrance:signWebToken");
+            data = await this.keyManager.signWebToken(message.args[0]);
+            console.log("HandleMessage exit:signWebToken", data);
+        } else if (message.method === "getPubKey") {
+            console.log("HandleMessage entrance:getPubKey");
+            data = await this.keyManager.getPubKey();
+            console.log("HandleMessage exit:getPubKey", data);
+        } else if (message.method === "deriveSharedSecret") {
+            console.log("HandleMessage entrance:deriveSharedSecret");
+            // @ts-ignore
+            data = await this.keyManager.deriveSharedSecret(message.args[0]);
+            console.log("HandleMessage exit:deriveSharedSecret", data);
+        } else if (message.method === "decryptMessage") {
+            console.log("HandleMessage entrance:decryptMessage");
+            // @ts-ignore
+            data = await this.keyManager.decryptMessage(message.args[0]);
+            console.log("HandleMessage exit:decryptMessage", data);
+        }
+        return {
+            data,
+            invocationId: message.invocationId,
+        };
+    }
+
+    private generateInvocationResponse(result: any) {
+        return {
+            invocationId: result.invocationId,
+            result,
+        };
+    }
+}
+
+export class RemoteKeyManager implements IKeyManager {
+    private remoteKeyClient: RemoteKeyClient;
+    private remoteUserId: CruxId;
+
+    constructor(cruxProtocolMessenger: CruxProtocolMessenger, remoteUserId: CruxId) {
+        this.remoteKeyClient = new RemoteKeyClient(cruxProtocolMessenger, remoteUserId);
+        this.remoteUserId = remoteUserId;
+    }
+
+    public async initialize() {
+        await this.remoteKeyClient.initialize();
+    }
+    // @ts-ignore
+    public signWebToken = async (token: any) => {
```

Comment:
Avoid so much code repitition. 
Just define what you need for invocation `"signWebToken", [token])` and call a common function

Reply:
working on it

Reply:
done